src/db.ts

```diff
@@ -36,17 +38,28 @@ export class DB {
     /** Postgres connection pool for primary database access. */
     postgres: Pool
     /** Redis used for various caches. */
-    redis: Redis.Redis
+    redisPool: GenericPool<Redis.Redis>
+
     /** Kafka producer used for syncing Postgres and ClickHouse person data. */
     kafkaProducer?: Producer
     /** ClickHouse used for syncing Postgres and ClickHouse person data. */
     clickhouse?: ClickHouse
 
     constructor(postgres: Pool, redis: Redis.Redis, kafkaProducer?: Producer, clickhouse?: ClickHouse) {
         this.postgres = postgres
-        this.redis = redis
         this.kafkaProducer = kafkaProducer
         this.clickhouse = clickhouse
+        this.redisPool = createPool<Redis.Redis>(
+            {
+                create: () => Promise.resolve(redis),
+                destroy: () => Promise.resolve(undefined),
```

Comment:
Should this not stop/close/disconnect the Redis client?

Reply:
Not if using just one client and the pool is effectively a glorified lock.